app/src/main/java/com/android/shelfLife/ui/recipes/Recipes.kt

```diff
@@ -0,0 +1,373 @@
+        } else {
+          recipeList.filter { recipe ->
+            recipe.name.contains(query, ignoreCase = true) // Filter by recipe name
+          }
+        }
+
+    if (selectedHousehold == null) {
+      FirstTimeWelcomeScreen(householdViewModel)
+    } else {
+      Scaffold(
+          modifier = Modifier,
+          topBar = {
+            selectedHousehold?.let {
+              TopNavigationBar(
+                  userHouseholds = householdViewModel.households.collectAsState().value,
+                  onHouseholdChange = { household ->
+                    if (household != selectedHousehold) {
+                      householdViewModel.selectHousehold(household)
+                    }
+                  },
+                  houseHold = it,
+                  householdViewModel = householdViewModel,
+                  onHamburgerClick = { scope.launch { drawerState.open() } },
+                  filters = filters)
+            }
+          },
+          bottomBar = {
+            BottomNavigationMenu(
+                onTabSelect = { destination -> navigationActions.navigateTo(destination) },
+                tabList = LIST_TOP_LEVEL_DESTINATION,
+                selectedItem = Route.RECIPES)
+          },
+          content = { paddingValues ->
+            Column(modifier = Modifier.padding(paddingValues).fillMaxSize()) {
+              RecipesSearchBar(query) { newQuery ->
+                query = newQuery // Update the query when user types
+              } // Pass query and update function to the search bar
+
+              // LazyColumn for displaying the list of filtered recipes
+              LazyColumn(modifier = Modifier.fillMaxSize()) {
+                items(filteredRecipes) { recipe ->
+                  RecipeItem(recipe, navigationActions, listRecipesViewModel)
+                }
+              }
+            }
+          })
+    }
+  }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+/**
+ * A composable function that displays a search bar for filtering recipes.
+ *
+ * @param query The current search query as a string.
+ * @param onQueryChange A callback function that gets triggered whenever the query changes, updating
+ *   the query state in the parent composable.
+ *
+ * This search bar provides a text input field where users can type to search for recipes. The UI
+ * layout consists of a box container that holds the search bar. It uses a `SearchBar` from the
+ * Material3 library, with options for managing its active state and search behavior.
+ * - The `query` parameter represents the current text in the search bar.
+ * - The `onQueryChange` function is invoked each time the user updates the search query, allowing
+ *   the parent composable to filter recipes based on user input.
+ * - The search bar displays a placeholder "Search recipes" when the query is empty.
+ * - The trailing icon (search icon) allows users to toggle the search bar's active state.
+ */
```

Comment:
Great explanation of this composable functionality, good job in providing good documentation! Keept this good habit!